activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb

```diff
@@ -22,8 +22,8 @@ def recreate_database(name, options = {}) #:nodoc:
         def create_database(name, options = {})
           options = { encoding: "utf8" }.merge!(options.symbolize_keys)
 
-          option_string = options.inject("") do |memo, (key, value)|
-            memo += case key
+          option_string = options.inject(+"") do |memo, (key, value)|
+            memo << case key
```

Comment:
`inject` shouldn't mutate memo; that's what `each_with_object` is for

Reply:
Actually.. is there a cop for that?

Reply:
AFAICS RuboCop has `Style/EachWithObject` cop, but it just check style, don't check the mutation.
https://rubocop.readthedocs.io/en/latest/cops_style/#styleeachwithobject

Actually I don't know we should avoid the mutation, since we already have `inject` (default by  Hash (`{}`) or Array (`[]`)) with the mutation in many places.

e.g.

https://github.com/rails/rails/blob/0bd4894d8b1977c00f4af64947ef860eee5d2b44/activesupport/lib/active_support/values/time_zone.rb#L271-L274

https://github.com/rails/rails/blob/82d184839e573b2085d3ea8007f81ab31827edd0/railties/lib/rails/source_annotation_extractor.rb#L128-L132

and more.